### DISCUSSION TITLE
Unit of work

### DISCUSSION THREAD

**Discussion by enjoykcc456** [mikro-orm/mikro-orm#6111] [General]
### Describe the bug

Hi, i did the following and expect process3 to be different to process or process2 as they are using different entity manager (default and forked - due to the use of transactional) which i assume they are using different identity map. But, the result shows that they are all the same. Hence, would like to check whats happening behind or am i doing something wrong.

Besides that, realise that even using the same entity manager to do findOne (the first and second findOne), there are 2 db queries instead of 1 showed in the console log. Would like to check is it expected? Because i am expecting only 1 db query with the unit of work. 

```typescript
    const process = await this.entityManager.findOne(ProcessEntity, '3376f684-a9a7-4fa4-a87a-e06ab6027fb3');
    const process2 = await this.entityManager.findOne(ProcessEntity, '3376f684-a9a7-4fa4-a87a-e06ab6027fb3');

    console.log({ result1: process === process2 });

    const process3 = await this.entityManager.transactional(async em => {
      const process3 = await em.findOne(ProcessEntity, '3376f684-a9a7-4fa4-a87a-e06ab6027fb3');
      return process3;
    });

    console.log({ result2: process === process3 });
```

Result: 
```console
[query] select "p0".* from "schemaA"."tableA" as "p0" where "p0"."id" = '3376f684-a9a7-4fa4-a87a-e06ab6027fb3' limit 1 [took 3 ms, 1 result]
[query] select "p0".* from "schemaA"."tableA" as "p0" where "p0"."id" = '3376f684-a9a7-4fa4-a87a-e06ab6027fb3' limit 1 [took 3 ms, 1 result]
{ result1: true }
[query] begin
[query] select "p0".* from "schemaA"."tableA" as "p0" where "p0"."id" = '3376f684-a9a7-4fa4-a87a-e06ab6027fb3' limit 1 [took 2 ms, 1 result]
[query] commit
{ result2: true }
```

### Reproduction

none

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

^6.3.9

### Node.js version

v18.16.0

### Operating system

MacOS 14.6.1 

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> Hi, i did the following and expect process3 to be different to process or process2 as they are using different entity manager (default and forked - due to the use of transactional) which i assume they are using different identity map. But, the result shows that they are all the same. Hence, would like to check whats happening behind or am i doing something wrong.

You call `em.transacitonal` on a fork with managed entities, so the fork it creates will have them available too, and those changes are propagated to the parent context automatically. So what you see is the correct behaviour.

> Besides that, realise that even using the same entity manager to do findOne (the first and second findOne), there are 2 db queries instead of 1 showed in the console log. Would like to check is it expected? Because i am expecting only 1 db query with the unit of work.

There should be just one query here, hard to say why there are two without seeing a complete reproduction.

**Reply by enjoykcc456** [reply]
Hi, ahh ok. Understand the first one.
For the second one I am unable to reproduce the issue again in my app, cant really recall what was done previously before the findOne calls. 
Thanks for the prompt reply!
